packages/zwave-js/src/lib/commandclass/EntryControlCC.ts

```diff
@@ -0,0 +1,683 @@
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 4);
+		this._sequenceNumber = this.payload[0];
+		this._dataType = this.payload[1] & 0b11;
+		this._eventType = this.payload[2];
+		const eventDataLength = this.payload[3];
+
+		validatePayload(this.payload.length >= 4 + eventDataLength);
+		this._eventData = Buffer.from(
+			this.payload.slice(4, 4 + eventDataLength),
+		);
+	}
+
+	private _sequenceNumber: number;
+	public get sequenceNumber(): number {
+		return this._sequenceNumber;
+	}
+
+	private _dataType: EntryControlDataTypes;
+	public get dataType(): EntryControlDataTypes {
+		return this._dataType;
+	}
+
+	private _eventType: EntryControlEventTypes;
+	public get eventType(): EntryControlEventTypes {
+		return this._eventType;
+	}
+
+	private _eventData: Buffer;
+	public get eventData(): Buffer {
+		return this._eventData;
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		const message: MessageRecord = {
+			"sequence number": this.sequenceNumber,
+			"data type": this.dataType,
+			"event type": this.eventType,
+			"event data": this.eventData.toString(),
+		};
+		return {
+			...super.toLogEntry(),
+			message,
+		};
+	}
+}
+
+@CCCommand(EntryControlCommand.KeySupportedReport)
+export class EntryControlCCKeySupportedReport extends EntryControlCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 1);
+		const length = this.payload[0];
+		validatePayload(this.payload.length >= 1 + length);
+		this._keySupported = parseBitMask(this.payload.slice(1, 1 + length), 0);
+		this.persistValues();
+	}
+
+	private _keySupported: number[];
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnly,
+		type: "number[]",
+		label: "Supported keys",
+		description: "A list of supported keys",
+	})
+	public get keySupported(): number[] {
+		return this._keySupported;
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		return {
+			...super.toLogEntry(),
+			message: { "supported keys": this.keySupported.toString() },
+		};
+	}
+}
+
+@CCCommand(EntryControlCommand.KeySupportedGet)
+@expectedCCResponse(EntryControlCCKeySupportedReport)
+export class EntryControlCCKeySupportedGet extends EntryControlCC {}
+
+@CCCommand(EntryControlCommand.EventSupportedReport)
+export class EntryControlCCEventSupportedReport extends EntryControlCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 6);
+
+		const dataTypeLength = this.payload[0] & 0b0000_0011;
+		validatePayload(this.payload.length >= 6 + dataTypeLength);
+		this._dataTypeSupported = parseBitMask(
+			this.payload.slice(1, 1 + dataTypeLength),
+			0,
+		);
+
+		const eventTypeLength = this.payload[1 + dataTypeLength] & 0b0001_1111;
+		validatePayload(
+			this.payload.length >= 6 + dataTypeLength + eventTypeLength,
```

Comment:
```suggestion
			this.payload.length >= 1 + dataTypeLength + eventTypeLength,
```
I prefer to work with an offset variable in cases where multiple variable-length fields are included in a command.
Otherwise you're going to repeat yourself a lot.